sqlair.go

```diff
@@ -50,7 +50,7 @@ var cacheMutex sync.RWMutex
 // It is ready to be run on a SQLair DB.
 type Statement struct {
 	cacheID stmtID
-	pe      *expr.PreparedExpr
+	te      *expr.TypedExpr
```

Comment:
What do you think about including a comments here saying we are caching the `TypeExpr` to be reused for all queries? It used to be that the `sqlair.Statement` had all the values necessary so it was self-explanatory but that is no longer that.

Reply:
I've added the comment:
```
// te is the type checked SQLair query. It is used to generate the required
// query values when the Statement is run on a database.e.
```

Reply:
Make it `type bounded` to be consistent with the terminology. I like the comments but I think we can better convey that we are only caching the tree. We really don't need this in the statement, we have all the information in the .Query and .Get args as well. And `it is used to generate` seems to imply that we are using this alone to generate the values when in reality we have to wait for the .Query step.

Reply:
I've changed it from type checked to type bound.

I'm avoiding using the word caching here to talk about the information stored in the type bound expression because the other field in this struct is called `cacheID` and refers to something else. I wouldn't say that we are simply caching the tree here. The SQL string only appears in the call to `Prepare` and we have added the type information, some of which only appears at the `.Get` stage and is needed at the `.Query` stage.

I've changed the comment to:
```
te is the type bound SQLair query. It contains information used to
generate query values from the input arguments when the Statement is 
run on a database.
```

Reply:
I think the new comment you suggested is perfect. And I agree we should not talk about caching in two different contexts which might get muddled.